### ISSUE TITLE
Sets of ordered sets are not equal to sets of sets

### ISSUE THREAD

**Issue by tsdh** [amalloy/ordered#16]
Ordered sets are equal to normal sets which is expected:

```clojure
(= (flatland.ordered.set/ordered-set 1 2 3) #{1 2 3})
;=> true
```

However, sets containing ordered sets aren't equal to sets containing sets:

```clojure
(= #{(flatland.ordered.set/ordered-set 1 2 3)} #{#{1 2 3}})
;=> false
```

I'm using `[org.flatland/ordered "1.5.2"]`.

**Comment by tsdh** [comment]
Oh, I'm sorry.  That seems to be a Clojure issue which can also be reproduced with other kinds of sets, e.g., `HashSet`.

**Comment by tsdh** [comment]
Clojure issue is here: http://dev.clojure.org/jira/browse/CLJ-1739

**Comment by jafingerhut** [comment]
This occurs in Clojure 1.6.0 and later because the implementation of hash changed from Clojure 1.5.1 to 1.6.0 for sets and maps, but the ordered-map implementation did not update then.  Here is a sample implementation in the data.avl library, which is updated to have compatible hash values with all versions of Clojure released: https://github.com/clojure/data.avl/blob/master/src/main/clojure/clojure/data/avl.clj#L53-L57

**Comment by amalloy** [comment]
Fixed in 1.5.3.

**Comment by tsdh** [comment]
Confirmed, thanks!
